nats.go

```diff
@@ -1683,6 +1700,70 @@ func (nc *Conn) createConn() (err error) {
 	return nil
 }
 
+// hasOCSPStatusRequest returns whether a certificate has the "status_request"
+// flag set.
+func hasOCSPStatusRequest(cert *x509.Certificate) bool {
+	tlsFeatures := asn1.ObjectIdentifier{1, 3, 6, 1, 5, 5, 7, 1, 24}
+	const statusRequestExt = byte(5)
+	const tlsFeaturesLen = 5
+
+	// Example Value: [48 3 2 1 5]
+	// Documentation:
+	// https://tools.ietf.org/html/rfc6066
+
+	for _, ext := range cert.Extensions {
+		if !ext.Id.Equal(tlsFeatures) {
+			continue
+		}
+		if len(ext.Value) != tlsFeaturesLen {
+			continue
+		}
+		return ext.Value[tlsFeaturesLen-1] == statusRequestExt
```

Comment:
This has the same problem as the similar logic in the nats-server PR, and needs the same fix of using `asn1.Unmarshal()`